File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] fin = new int[nums.length];

        fin[0] = 1;
        for (int i = 1; i < nums.length; i++) {
            fin[i] = fin[i - 1] * nums[i - 1];
        }

        int post = 1;
        for (int i = nums.length - 1; i >= 0; i--) {
            fin[i] = fin[i] * post;
            post *= nums[i];
        }

        return fin;
    }
}  

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        sub = []
        def dfs(i, r):
            if i >= len(nums):
                return
            
            if r == 0:
                res.append(sub.copy())
                return 

            if r < 0:
                return  

            sub.append(nums[i])
            dfs(i, r - nums[i])
            sub.pop()

            dfs(i + 1, r)

        dfs(0, target)
        return res    